docs/couler-tekton-design.md

```diff
@@ -0,0 +1,70 @@
+# Couler Tekton Design
+
+This document outlines the design of tekton backend for core Couler APIs.
+
+## Goals
+
+* Design the tekton backend implementation for core Couler APIs.
+
+## Non-Goals
+
+* Provide implementation details.
+
+## Design
+
+Based on the core [Couler APIs design](https://github.com/couler-proj/couler/pull/24), we can see the definitions in different workflow engines.
```

Comment:
This PR has been merged so you can point this to https://github.com/couler-proj/couler/blob/master/docs/couler-api-design.md instead.